### ISSUE TITLE
The number of return values of `framework.NewPodInfo` in kubernetes / kubernetes has changed

### ISSUE THREAD

**Issue by AxeZhan** [kubernetes-sigs/scheduler-plugins#436]
After [112219](https://github.com/kubernetes/kubernetes/pull/112219) got merged. We have to change our way to call `framework.NewPodInfo`.

**Before**
```
func NewPodInfo(pod *v1.Pod) *PodInfo {
	pInfo := &PodInfo{}
	pInfo.Update(pod)
	return pInfo
}
```

**now**
```
func NewPodInfo(pod *v1.Pod) (*PodInfo, error) {
	pInfo := &PodInfo{}
	err := pInfo.Update(pod)
	return pInfo, err
}

```

I think currently we only use this function in `pkg/coscheduling/coscheduling_test.go`. So this should be a small change.
I'll create a new function and relace old function with it in test files. Like I did in kubernetes repo.
```
 func mustNewPodInfo(t *testing.T, pod *v1.Pod) *framework.PodInfo {
     t.Helper()
     podInfo, err := framework.NewPodInfo(pod)
     if err != nil {
         t.Fatal(err)
     }
     return podInfo
 }
```
I'll be doing this when `k8s.io/kubernetes` v1.26 releases.

My concern here is: will this behavior(updating k8s deps from v1.24.3 to v.1.26) brings other problems?

**Comment by Huang-Wei** [comment]
> My concern here is: will this behavior(updating k8s deps from v1.24.3 to v.1.26) brings other problems?

I'm not sure I understand this question. Basically we bump the k8s deps for every release, so right now `master` is about to bump to 1.25 #466.

So we just wait for a couple of months, and then bump to a new release as usual. BTW: I tag this issue to milestone `v1.26`.

**Comment by Huang-Wei** [comment]
This was fixed by #489.
